addons/web_tour/static/src/tour_service/tour_interactive.js

```diff
@@ -364,18 +388,15 @@ export class TourInteractive {
 
     _onMutation() {
         if (this.currentAction) {
-            let tempAnchor = hoot.queryFirst(this.currentAction.anchor, { visible: true });
-            tempAnchor = tempAnchor && this.getAnchorEl(tempAnchor, this.currentAction.event);
+            const tempAnchors = this.findTriggers();
             if (
-                (!this.anchorEl && tempAnchor) ||
-                (this.anchorEl && tempAnchor && tempAnchor !== this.anchorEl)
+                (!this.anchorEl && tempAnchors.length) ||
+                (this.anchorEl && tempAnchors.length && !tempAnchors.includes(this.anchorEl))
```

Comment:
Bit of nitpicking but this could be simplified like this:
```js
tempAnchors.length && (!this.anchorEl || !tempAnchors.includes(this.anchorEl)
```

Reply:
voila !